src/lib/snark_work_lib/metrics.ml

```diff
@@ -116,7 +68,12 @@ let emit_partitioned_metrics (result_with_spec : _ Partitioned_spec.Poly.t) =
           Counter.inc zkapp_proof_updates
             (match spec with Proved -> 1.0 | _ -> 0.0) ;
           Counter.inc snark_work_zkapp_base_time_sec (Time.Span.to_sec elapsed))) ;
-      Sub_zkapp_snark_generated { kind = `Segment; elapsed }
+
+      [%log info] "Sub-zkApp SNARK generated in $elapsed of type $kind"
+        ~metadata:
+          [ ("elapsed", `Float (Time.Span.to_sec elapsed))
+          ; ("kind", `String "Segment")
```

Comment:
Would be great to include authorization kind in the logging metadata: `Proved`, `Signed_single`, `Signed_pair`. Let's derive json for `Zkapp_command.possible_segments` and use it here.